src/main/java/fi/metatavu/dcfb/server/stripe/StripeWebhookHandler.java

```diff
@@ -0,0 +1,120 @@
+package fi.metatavu.dcfb.server.stripe;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+
+import com.stripe.exception.SignatureVerificationException;
+import com.stripe.model.Charge;
+import com.stripe.model.Event;
+import com.stripe.model.StripeObject;
+import com.stripe.net.ApiResource;
+import com.stripe.net.Webhook;
+
+import fi.metatavu.dcfb.server.items.ItemController;
+import fi.metatavu.dcfb.server.persistence.model.Item;
+import fi.metatavu.dcfb.server.persistence.model.ItemReservation;
+import fi.metatavu.dcfb.server.settings.SystemSettingController;
+import fi.metatavu.dcfb.server.webhooks.WebhookException;
+import fi.metatavu.dcfb.server.webhooks.WebhookHandler;
+
+/**
+ * Class that handles webhooks from stripe
+ * 
+ * @author Heikki Kurhinen
+ */
+@RequestScoped
+public class StripeWebhookHandler implements WebhookHandler {
+  
+  @Inject
+  private ItemController itemController;
+
+  @Inject
+  private SystemSettingController systemSettingController;
+
+  @Inject
+  private Logger logger;
+
+  @Override
+  public String getType() {
+    return StripeConsts.STRIPE_WEBHOOK_TYPE;
+  }
+
+  @Override
+  public void handle(HttpServletRequest request) throws WebhookException {
+    String payload = null;
+    String signatureHeader = request.getHeader(StripeConsts.STRIPE_SIGNATURE_HEADER);
+    try {
+      payload = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+    } catch (IOException e) {
+      throw new WebhookException(e);
+    }
+
+    Event event = null;
+    try {
+      event = Webhook.constructEvent(payload, signatureHeader, systemSettingController.getSettingValue(StripeConsts.STRIPE_SIGNING_KEY_SETTING));
+    } catch (SignatureVerificationException e) {
+      throw new WebhookException(e);
+    }
+
+    if (inLiveMode() != event.getLivemode()) {
+      throw new WebhookException("Operation mode mismatch");
+    }
+
+    switch(event.getType()) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this "switch" statement by "if" statements to increase readability. ([squid:S1301](https://sonarcloud.io/organizations/metatavu-oy/rules#rule_key=squid%3AS1301))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fi.metatavu.dcfb-api%3Adcfb-api&pullRequest=18&issues=AWVh4kV4Uj60SLv4Ylyq&open=AWVh4kV4Uj60SLv4Ylyq)

[ISSUE_KEY]: AWVh4kV4Uj60SLv4Ylyq